big_tests/tests/connect_SUITE.erl

```diff
@@ -546,6 +571,52 @@ bind_server_generated_resource(Config) ->
     ?assert(is_binary(Resource)),
     ?assert(byte_size(Resource) > 0).
 
+same_resource_replaces_session(Config) ->
+    UserSpec = [{resource, <<"conflict">>} | escalus_users:get_userspec(Config, alice)],
+    {ok, Alice1, _} = escalus_connection:start(UserSpec),
+
+    {ok, Alice2, _} = escalus_connection:start(UserSpec),
+
+    ConflictError = escalus:wait_for_stanza(Alice1),
+    escalus:assert(is_stream_error, [<<"conflict">>, <<>>], ConflictError),
+
+    mongoose_helper:wait_until(fun() -> escalus_connection:is_connected(Alice1) end, false),
+
+    escalus_connection:stop(Alice2).
+
+clean_close_of_replaced_session(Config) ->
+    lager_ct_backend:capture(warning),
+
+    same_resource_replaces_session(Config),
+
+    lager_ct_backend:stop_capture(),
+    FilterFun = fun(_, Msg) ->
+                        re:run(Msg, "replaced_wait_timeout") /= nomatch
+                end,
+    [] = lager_ct_backend:recv(FilterFun).
+
+replaced_session_cannot_terminate(Config) ->
+    % GIVEN a session that is frozen and cannot terminate
+    lager_ct_backend:capture(warning),
+    UserSpec = [{resource, <<"conflict">>} | escalus_users:get_userspec(Config, alice)],
+    {ok, _Alice1, _} = escalus_connection:start(UserSpec),
+    [C2SPid] = children_specs_to_pids(rpc(mim(), supervisor, which_children, [ejabberd_c2s_sup])),
+    ok = rpc(mim(), sys, suspend, [C2SPid]),
+
+    % WHEN a session gets replaced ...
+    {ok, Alice2, _} = escalus_connection:start(UserSpec),
+
+    % THEN a timeout warning is logged
+    FilterFun = fun(_, Msg) ->
```

Comment:
Consider extracting this lambda to named private method because you use it in `clean_close_of_replaced_session/1` and `replaced_session_cannot_terminate/1` ??